shared/gateway/gateway_test.go

```diff
@@ -0,0 +1,86 @@
+package gateway
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/cmd/beacon-chain/flags"
+	"github.com/prysmaticlabs/prysm/shared/testutil/require"
+	logTest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/urfave/cli/v2"
+)
+
+// Test that beacon gateway Start, Stop.
+func TestBeaconGateway_StartStop(t *testing.T) {
+	hook := logTest.NewGlobal()
+
+	app := cli.App{}
+	set := flag.NewFlagSet("test", 0)
+	ctx := cli.NewContext(&app, set, nil)
```

Comment:
I can't see why you need all these variables. Why can't you pass values directly into `NewBeacon`?

Reply:
```
beaconGateway := NewBeacon(
		ctx.Context,
		BeaconCaller,
		selfAddress,
		selfCert,
		gatewayAddress,
		nil, /*optional mux*/
		allowedOrigins,
		enableDebugRPCEndpoints,
		ctx.Uint64("grpc-max-msg-size"),
	)
```

Will be 

```
beaconGateway := NewBeacon(
		cli.NewContext(&cli.App{}, flag.NewFlagSet("test", 0), nil).Context,
		BeaconCaller,
		selfAddress,
		selfCert,
		gatewayAddress,
		nil, /*optional mux*/
		allowedOrigins,
		enableDebugRPCEndpoints,
		ctx.Uint64("grpc-max-msg-size"),
	)
```

Reply:
The usual way create a context in tests is through `ctx := context.Background()`.

Reply:
The thing is the cli library allows for the fetching of values from the flags efficiently. 
e.g. 
`gatewayPort := ctx.Int(flags.GRPCGatewayPort.Name)`

will be something like  this ; done directly which albeit being intuitive was not the way currently used in the code. 
`gatwayPort :=  flags.GRPCGatewayPort.Value `